src/table_compaction.cc

```diff
@@ -376,7 +376,7 @@ Status TableMgr::compactLevelItr(const CompactOptions& options,
         //   1) number of files after split, and
         //   2) min keys for each new file.
         do {
-            if (!isCompactionAllowed()) {
+            if (!isCompactionAllowed() && !adjusting_num_l0) {
```

Comment:
This is not acceptable. No matter what the reason is, compaction should be cancelled upon the close of DB instance.